### PR TITLE
Add option to expose a filterproxy

### PR DESCRIPTION
**Pull request by glrf**
With this we can forward a view of the https://github.com/vshn/exporter-filterproxy. Is used to provide the metrics related to the namespace.

#### Checklist

- [x] Chart Version bumped
- [x] I have run `make docs`
- [x] Variables are documented in the values.yaml using the format required by [Helm-Docs](https://github.com/norwoodj/helm-docs#valuesyaml-metadata).
- [x] PR contains the label that identifies the chart, e.g. `chart/<chart-name>`
- [x] PR contains the label that identifies the type of change, which is one of
      [ `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency` ]

### CODE REVIEW

**Review by simu — APPROVED** [review]
LGTM overall

**Review by TheBigLee — APPROVED** [review]
LGTM

File: appuio/haproxy/README.gotmpl.md [hunk 1]
```diff
@@ -134,17 +134,23 @@ Set `haproxy.config` to `redisk8s` to use the Redis configuration with DNS servi
...
-* `<haproxy_service_ip>:9090/metrics/mariadb-1/metrics`
+* `<haproxy_service_ip>:9090/mariadb/0`
+* `<haproxy_service_ip>:9090/mariadb/1`
 * `...`
 
 For Redis the endpoints are:
-* `<haproxy_service_ip>:9090/metrics/redis-0/metrics`
-* `<haproxy_service_ip>:9090/metrics/redis-1/metrics`
+* `<haproxy_service_ip>:9090/redis/0`
+* `<haproxy_service_ip>:9090/redis/1`
```
Comment (anothertobi) [inline appuio/haproxy/README.gotmpl.md:155]: The new paths are correct for the filterproxy implementation. The regular metrics path would actually be `<haproxy_service_ip>:9090/metrics` with a host header of `redis-0`, `redis-1`, etc.

```
curl -H "Host: redis-0" haproxy:9090/metrics
```

**Review by anothertobi — CHANGES_REQUESTED** [review]
My understanding is that the exporter-filterproxy would return all metrics filtered for the given namespace and not the metrics for a specific backend node.

Example:
A request to `haproxy:9090/mariadb/0` would return all metrics for the namespace the HAProxy is running in and not just the metrics of the backend `galera-node-0` which would not require any namespace restriction.

Therefore, a single endpoint (on a different port) seems more appropriate.

File: appuio/haproxy/README.gotmpl.md (hunk 1 above)
Comment (glrf) [reply appuio/haproxy/README.gotmpl.md:155]: You're right this doesn't explain the whole picture (and before the documentation was completely wrong though) I'll improve it

**Comment by glrf** [comment]
> My understanding is that the exporter-filterproxy would return all metrics filtered for the given namespace and not the metrics for a specific backend node.
> 
> Example: A request to `haproxy:9090/mariadb/0` would return all metrics for the namespace the HAProxy is running in and not just the metrics of the backend `galera-node-0` which would not require any namespace restriction.

No `haproxy:9090/mariadb/0` won't be forwarded to the filterproxy at all but to the first mariadb instance. Also the filerproxy doesn't return all metrics for a namespace but can proxy specific exporters and filter them by label (in this case by namespace and in the first iteration will be used to expose kube-state-metrics only)

**Review by anothertobi — APPROVED** [review]
The new `/metrics` shortcut had me confused. LGTM 👍

**Pull request merged**